pkg/controller/erspan.go

```diff
@@ -0,0 +1,428 @@
+	cont.initErspanInformerBase(
+		cache.NewListWatchFromClient(
+			erspanClient.AciV1alpha().RESTClient(), "erspanpolicies",
+			metav1.NamespaceAll, fields.Everything()))
+}
+
+func (cont *AciController) initPodIfInformerFromClient(
+	podIfClient *podIfclientset.Clientset) {
+	cont.initPodIfInformerBase(
+		cache.NewListWatchFromClient(
+			podIfClient.AciV1().RESTClient(), "podifs",
+			metav1.NamespaceAll, fields.Everything()))
+}
+
+func (cont *AciController) initErspanInformerBase(listWatch *cache.ListWatch) {
+	cont.erspanIndexer, cont.erspanInformer = cache.NewIndexerInformer(
+		listWatch, &erspanpolicy.ErspanPolicy{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.erspanPolicyUpdated(obj)
+			},
+			UpdateFunc: func(_, obj interface{}) {
+				cont.erspanPolicyUpdated(obj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.erspanPolicyDeleted(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing erspan Policy Informers")
+}
+
+func (cont *AciController) initPodIfInformerBase(listWatch *cache.ListWatch) {
+	cont.podIfIndexer, cont.podIfInformer = cache.NewIndexerInformer(
+		listWatch, &podIfpolicy.PodIF{}, 0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cont.podIFAdded(obj)
+			},
+			UpdateFunc: func(oldobj interface{}, newobj interface{}) {
+				cont.podIFAdded(newobj)
+			},
+			DeleteFunc: func(obj interface{}) {
+				cont.podIFDeleted(obj)
+			},
+		},
+		cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc},
+	)
+	cont.log.Debug("Initializing podif Informers")
+}
+
+func getPodifKey(podif *podIfpolicy.PodIF) string {
+	return podif.Status.PodNS + "/" + podif.Status.PodName
+}
+
+func isValidEPG(podif *podIfpolicy.PodIF) bool {
+	pattern := regexp.MustCompile("\\|")
+	idx := pattern.FindAllStringIndex(podif.Status.EPG, -1)
+	if len(idx) != 1 || idx[0][0] == 0 || idx[0][0] == len(podif.Status.EPG)-1 {
+		return false
+	}
+	return true
+}
+
+func (cont *AciController) podIFAdded(obj interface{}) {
+	podif := obj.(*podIfpolicy.PodIF)
+	cont.log.Infof("podif Added: %s", podif.ObjectMeta.Name)
+	cont.indexMutex.Lock()
+	if isValidEPG(podif) {
+		deconcatenate := strings.Split(podif.Status.EPG, "|")
+		deconEPG := deconcatenate[len(deconcatenate)-1]
+		deconAppProfile := deconcatenate[0]
+		podifdata := &EndPointData{
+			MacAddr:    podif.Status.MacAddr,
+			EPG:        deconEPG,
+			Namespace:  podif.Status.PodNS,
+			AppProfile: deconAppProfile,
+		}
+		podifKey := getPodifKey(podif)
+		cont.podIftoEp[podifKey] = podifdata
+	} else {
+		cont.log.Infof("Invalid EPG: %s", podif.Status.EPG)
+	}
+	cont.indexMutex.Unlock()
+}
+
+func (cont *AciController) podIFDeleted(obj interface{}) {
+	podif := obj.(*podIfpolicy.PodIF)
+	cont.log.Infof("podif Deleted: %s", podif.ObjectMeta.Name)
+	cont.indexMutex.Lock()
+	delete(cont.podIftoEp, podif.ObjectMeta.Name)
```

Comment:
This will leak here Key should be getPodifKey(podif)

Reply:
Good catch, will fix it